sky/packages/sky/lib/src/material/bottom_sheet.dart

```diff
@@ -101,9 +133,17 @@ class _ModalBottomSheetRoute extends PerformanceRoute {
     );
   }
 
+  void didPush(NavigatorState navigator) {
+    super.didPush(navigator);
+    _performance?.forward();
+  }
+
   void didPop([dynamic result]) {
     completer.complete(result);
-    super.didPop(result);
+    if (_performance.status != PerformanceStatus.dismissed) {
+      _performance?.reverse();
+      super.didPop(result);
```

Comment:
My mistake. It was PerformanceRoute.didPop() that caused problems for me because it unconditionally calls performance.reverse().  I'm not subclassing PerformanceRoute anymore so it's no longer an issue. I've changed the code to unconditionally call super.didPop().